### PR TITLE
[prometheusreceiver] enable reporting of additional metrics issue #21064

### PR DESCRIPTION
**Pull request by premendrasingh**
Feature: Enable reporting of additional metrics for Prometheus client used in PrometheusReceiver

**Description:**
Prometheus client has a featureFlag options.ExtraMetrics which reports content size metric ([scrape_body_size_bytes](https://github.com/prometheus/prometheus/blob/main/scrape/scrape.go#L1770)) for successful scrapes and -1 for dropped data. This metric will help us collect data and setup alert for dropped data. However the prometheus receiver does not have a way to enable this feature flag.

This PR passes the flag to scrape.NewManager() to enable reporting of metric - [scrape_body_size_bytes](https://github.com/prometheus/prometheus/blob/main/scrape/scrape.go#L1770)

**Link to tracking Issue:**
#21040 

**Testing:**
Added check for validating the value set in testdata/config.yaml

### CODE REVIEW

**Comment by premendrasingh** [comment]
@JaredTan95, @dashpole 
CLA is signed, please help review this PR. 
Thanks.

**Comment by JaredTan95** [comment]
thx your contribution, But could you add some tests to cover your enhanced or the scraped results if `report_extra_metrics` is true. :-P

**Review by JaredTan95 — CHANGES_REQUESTED** [review]
thx your contribution, But could you add some tests to cover your enhanced or the scraped result's diffs when `report_extra_metrics` is true or false?  :-P

File: receiver/prometheusreceiver/testdata/config.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ prometheus/customname:
   buffer_count: 45
   use_start_time_metric: true
   start_time_metric_regex: '^(.+_)*process_start_time_seconds$'
+  report_extra_metrics: true
```
Comment (dashpole) [inline receiver/prometheusreceiver/testdata/config.yaml:7]: I would suggest including "scrape" in the name, as it is a scrape.Option.  Possibly "report_extra_scrape_metrics".

**Review by dashpole — COMMENTED** [review]
lgtm overall.  Just one suggestion

**Comment by premendrasingh** [comment]
Thanks @JaredTan95 @dashpole for the suggestions. Added test case and changed name to `report_extra_scrape_metrics`

**Review by dashpole — APPROVED** [review]

**Comment by premendrasingh** [comment]
@JaredTan95  @Aneurysm9 @dashpole I have added change log. Please help review. Thanks.

**Comment by JaredTan95** [comment]
> @JaredTan95 @Aneurysm9 @dashpole I have added change log. Please help review. Thanks.

great, and last thing, we should update the `README.md` docs for describing `report_extra_scrape_metrics` .

**Comment by premendrasingh** [comment]
@JaredTan95  Updated README.md

File: receiver/prometheusreceiver/README.md [hunk 2]
```diff
@@ -69,6 +69,8 @@ prometheus --config.file=prom.yaml
 "--feature-gates=receiver.prometheusreceiver.UseCreatedMetric"
 ```
 
+- `receiver.prometheusreceiver.report_extra_scrape_metrics`: Extra prometheus scrape metrics can be reported by setting this parameter to `true`
```
Comment (JaredTan95) [inline receiver/prometheusreceiver/README.md:72]:
```suggestion
- `report_extra_scrape_metrics`: Extra Prometheus scrape metrics can be reported by setting this parameter to `true`
```

Comment (premendrasingh) [reply receiver/prometheusreceiver/README.md]: Thanks for the suggestion.

**Review by JaredTan95 — APPROVED** [review]
LGTM

@Aneurysm9 pls take a look

**Comment by premendrasingh** [comment]
Thanks @JaredTan95, @dashpole, @mx-psi for helping with the PR merge process.

**Pull request merged**